flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -172,6 +229,15 @@ public AsyncSinkWriter(
 
         this.inFlightRequestsCount = 0;
         this.bufferedRequestEntriesTotalSizeInBytes = 0;
+
+        this.metrics = context.metricGroup();
+        this.metrics.setCurrentSendTimeGauge(
+                () -> {
+                    long time = this.ackTime - this.lastSendTimestamp;
+                    return time < 0 ? this.lastSendDuration : time;
```

Comment:
Is this thread safe when there are multiple in-flight requests? How do we know `ackTime` and `lastSendTimestamp` are the same request?

Reply:
Is this the same as `lastSendDuration`?

Reply:
It's true, race conditions between the in-flight requests occurs. I've modified it such that updates to `ackTime` and `lastSendTimestamp` occur in the same single threaded block now. `lastSendDuration` has been removed since it's not necessary. Added a test to prove too.